### ISSUE TITLE
Clarification on Recent Changes to Loss and Gradient Accumulation

### ISSUE THREAD

**Issue by jiosephlee** [huggingface/transformers#39567]
Hi!

I've been loosely following the recent conversations on bugs/issues such as https://github.com/huggingface/transformers/pull/34198 https://github.com/huggingface/transformers/pull/34191. As a lay user, it's not entirely clear to me what the issue is.

To hone in on specific questions, as someone who wants to use Trainer with a custom loss function, I'm concerned that there are numerous factors I need to account for, which leads to the following questions:
1. When providing a custom `compute_loss_func`, Is the expectation to divide the loss by num_items_in_batch (batch_size * gradient accumulation)?
2. Is the gradient accumulation handled somewhere else? My understanding is that we are pre-fetching the number of samples seen due to accumulation via num_items_in_batch, and so this loss is divided in advance but later added up..?
3. I'm seeing open issues, such as https://github.com/huggingface/transformers/issues/38837, which appear to be unique to the last step when the gradient accumulation isn't nicely divided. Is this still an issue?
4. Do any of these dynamics change in a multi-GPU setup?

**Comment by Rocketknight1** [comment]
This isn't really a bug - try asking in our [support space](https://huggingface.co/spaces/transformers-community/support) instead!

**Comment by RonanKMcGovern** [comment]
Thanks, I've the same question - seems unanswered in the forum too: https://huggingface.co/spaces/transformers-community/support/discussions/15
